### PR TITLE
Move connection error vs close handling more towards correct

### PR DESCRIPTION
**Pull request by jcorbin**
Had to back off from making close event node-idiomatic (take only a "was it a clean close" boolean) due to how coupled we are currently to getting the error through close.

r @raynos @kriskowal

### CODE REVIEW

File: node/connection.js [hunk 1]
```diff
@@ -554,16 +537,35 @@ TChannelConnection.prototype.resetAll = function resetAll(err) {
...
             outArg1: String(req.arg1)
         };
-        if (err.type === 'tchannel.socket-local-closed') {
-            err = errors.TChannelLocalResetError(err, info);
+
+        var reqErr = err;
+        if (reqErr.type === 'tchannel.socket-local-closed') {
+            reqErr = errors.TChannelLocalResetError(reqErr, info);
         } else {
-            err = errors.TChannelConnectionResetError(err, info);
```
Comment (kriskowal) [inline node/connection.js]: sonofa

**Comment by kriskowal** [comment]
:ok:

File: node/peer.js [hunk 2]
```diff
@@ -205,30 +205,35 @@ function waitForIdentified(callback) {
...
     if (conn.closing) {
         callback(conn.closeError);
-    } else if (conn.remoteName) {
+        return;
+    }
+
+    if (conn.remoteName) {
         callback(null);
-    } else {
-        self._waitForIdentified(conn, callback);
```
Comment (Raynos) [inline node/peer.js]: There was a reason for this.

I do not want to allocate closures in `waitForIdentified()` and moved the closure allocation into a seperate function.

**Comment by Raynos** [comment]
Let's add the `_waitForIdentified()` method back.

Other then that lgtm.

**Comment by jcorbin** [comment]
added back, travis.then(merge)

**Pull request merged**